interfaces/cython/cantera/examples/onedim/UnscaleVariances.py

```diff
@@ -0,0 +1,655 @@
+    def _ReadHeader(self):
+        count = 1
+        done = False
+        temp = self._input.readline()
+        while done == False:
+            name = temp.strip()
+            if name == '[NUMBER_FMEAN]':
+                temp = self._input.readline().strip()
+                self._numGridpoints = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_CPROGRESS]':
+                temp = self._input.readline().strip()
+                self._numFlamelets = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_FVAR]':
+                temp = self._input.readline().strip()
+                self._numFvars = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_CVAR]':
+                temp = self._input.readline().strip()
+                self._numCvars = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_SPECIES]':
+                temp = self._input.readline().strip()
+                self._numSpecies = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_QOI]':
+                temp = self._input.readline().strip()
+                self._numQoi = int(temp)
+                count = count + 1
+            elif name == '[NUMBER_VARIABLES]':
+                temp = self._input.readline().strip()
+                self._numVars = int(temp)
+                count = count + 1
+            elif name == '[DATA]':
+                temp = self._input.readline().strip()
+                self._columnHeader = temp
+                done = True
+                count = count + 1
+            if done == False:
+                temp = self._input.readline()
+                count = count + 1
+        return count
+
+    # read in the input file header
+    def Read(self):
+
+        # read in the header and the data
+        lineCount = self._ReadHeader()
+
+        arrayLen = self._numCvars * self._numFvars * self._numFlamelets * self._numGridpoints
+        arrayWidth = self._numVars
+        self._data = np.zeros((arrayLen, arrayWidth), dtype=np.double)
+
+        for i, line in enumerate(self._input):
+            for j, temp in enumerate(line.split()):
+                self._data[i,j] = float(temp)
+
+        # close the file
+        self._input.close()
+
+# OutputFileHandler class - responsible for writing output file 
+class OutputFileHandler():
+
+    def __init__(self, fileName):
+        self._fileName = fileName
+        self._pressure = 0          # pressure
+        self._numGridpoints = 0     # num fmean
+        self._numFlamelets = 0      # num cmean
+        self._numFvars = 0          # num fvar 
+        self._numVars = 0           # num variables 
+        self._numCvars = 1          # num of cvar
+        self._numSpecies = 0        # num of species
+        self._numQoi = 0            # num QOI
+        self._columnHeader = "none" # column headings
+        self._output = open(fileName,'w')

```

Comment:
Same comment here about file handles. All of the output can be collected in, e.g., a list and then written to a file in a single context block.